packages/nx/src/command-line/release/config/config.ts

```diff
@@ -694,3 +704,41 @@ function hasInvalidGitConfig(
     !!userConfig.git && !!(userConfig.version?.git || userConfig.changelog?.git)
   );
 }
+
+async function getDefaultProjects(
+  projectGraph: ProjectGraph,
+  projectFileMap: ProjectFileMap
+): Promise<string[]> {
+  // default to all library projects in the workspace with a package.json file
+  return findMatchingProjects(['*'], projectGraph.nodes).filter(
+    (project) =>
+      projectGraph.nodes[project].type === 'lib' &&
+      // Exclude all projects with "private": true in their package.json because this is
+      // a common indicator that a project is not intended for release.
+      // Users can override this behavior by explicitly defining the projects they want to release.
+      isProjectPublic(project, projectGraph, projectFileMap)
+  );
+}
+
+function isProjectPublic(
```

Comment:
Future Work: I wonder if there's a good way to abstract this part out into something like the project metadata that @FrozenPandaz and I are about to work on